### ISSUE TITLE
Unable to use Live Deploy on Android

### ISSUE THREAD

**Issue by diegozuber** [ionic-team/cordova-plugin-ionic#107]
I've packaged the same build, that works now on iOS perfectly for Android with Ionic Package. if I start the App it downloads the Update as expected.

`IonicCordova.deploy.delegate.then(console.log, console.error)` Output before redirect:
![bildschirmfoto 2018-07-27 um 10 55 50 am](https://user-images.githubusercontent.com/21689209/43312375-5e319644-918d-11e8-8bba-280b827b312d.png)

After I want to redirect to the new version I get just a gray screen and after a while a debug message appears in a debug build.

![screenshot_2018-07-27-10-56-22-953_ch winsun evagruen](https://user-images.githubusercontent.com/21689209/43312405-7bc7f82e-918d-11e8-9d6b-5cb61a78af56.png)

After I click okay on that message, the app quits. Then if I restart the App the splash screen is shown and never disappear. I've got the following in the console:
![bildschirmfoto 2018-07-27 um 10 56 44 am](https://user-images.githubusercontent.com/21689209/43312412-7f97eb12-918d-11e8-97e3-7bf64555eef2.png)

`IonicCordova.deploy.delegate.then(console.log, console.error)` shows the following:
![bildschirmfoto 2018-07-27 um 10 57 02 am](https://user-images.githubusercontent.com/21689209/43312419-84a3a100-918d-11e8-8b26-05dddc3df899.png)

I'm testing on a Xiaomi Redmi Note 4 with Android 6.0. I'm using these plugin Versions:
```
cordova-plugin-ionic v5.0.2
cordova-plugin-ionic-webview v2.0.1
@ionic/pro v2.0.1
cordova-android v7.0.0
```

**Comment by diegozuber** [comment]
Update to `cordova-android v7.1.1` didn't make any difference.

**Comment by nphyatt** [comment]
Thanks for all the details @diegozuber! That should be enough to help me reproduce. I’ll look into it today.

**Comment by nphyatt** [comment]
@diegozuber I think I found the issue. See (https://github.com/ionic-team/cordova-plugin-ionic/pull/108) I've cut a release `5.0.3` can you try upgrading and verifying it fixes the issue?

Thanks!

**Comment by diegozuber** [comment]
@nphyatt Thanks for the new version. Unfortunately the new version 5.0.3 didn't fix the problem.

**Comment by nphyatt** [comment]
@diegozuber did you at least stop seeing the error with `Failed to construct URL` ? I think we may be dealing with 2 seperate issues here....I hope the fix addressed the URL error but this sounds like it could also be related to #109

**Comment by diegozuber** [comment]
@nphyatt Thanks for you answer. The current behavior is as follows:

(I'm still using a custom implementation of the update process, UPDATE_METHOD=none)

1. I build the apk with Ionic Deploy
2. Installed it on the device (xiaomi redmi note 4 or I tested it also on a Honor 7)
3. I open the app and it downloads the update as expected
4. then I redirect it to the new version (before the redirect I show the splash screen for UX)
5. after a while the splash screen disappears and the gray screen appears with the error message from above

**NEW behavior since v5.0.2**
As soon as I restart the app after step 5) it shows the splash screen and then it crashes. See the Bug Report:
![screenshot_2018-07-30-15-08-06-460_com miui bugreport](https://user-images.githubusercontent.com/21689209/43399592-be4cf4da-940b-11e8-8e2f-82ec26b70bc1.png)

**NOTE: If I build the same commit for iOS, all works fine on an iPhone X**

**Comment by nphyatt** [comment]
@diegozuber ok I think we identified that it's actually a bug in the splash screen plugin causing the issue. The way the deploy plugin works just exposes the issue. Looks like someone opened a PR with apache (maintainers of the plugin) to fix the issue back in January but it has never been merged. We forked a version and added that fix to see if it would help and as far as we can tell it does.

We're looking into whether apache will merge the PR into the plugin or if we should fork it and maintain a version for ourselves. In the meantime would could you try doing the following to test that it does indeed fix the issue for you?

```
cordova plugin rm cordova-plugin-splashcreen
cordova plugin add https://github.com/nphyatt/cordova-plugin-splashscreen.git
```

**Comment by diegozuber** [comment]
@nphyatt Thanks a lot for your investigation. Unfortunately I'm still getting the same errors.

- after the initial app start & update I get a gray screen with the message as in my very first comment in this issue
- and after I restart the app it crashes (since v5.0.3) and again the same splash screen null pointer exception.

How can I help you to find the bug?

**Comment by nphyatt** [comment]
Thanks for all the reporting help and patience @diegozuber. 
I was able to look at your app and dig in a little deeper. There were a multitude of issues going on causing debugging to be very hard. 

1. I believe the gray screen with the message was being caused by an issue with cordova-plugin-ionic-webview if you install the version 2.0.2 I think it will fix that.

2. The fix I suggested earlier with installing my forked version of the `cordova-plugin-splashscreen` will fix the null pointer however I got the command slightly wrong...apparently you need to add with `cordova plugin add https://github.com/nphyatt/cordova-plugin-splashscreen.git --save`. Apparently if you don't use the `--save` flag cordova doesn't update the package.json and so when pro installs the plugin for you're package it reverts to using the official one.

3. All of this is made more confusing by a different bug I found in the deploy plugin where if you have downloaded and cached an update on a previous binary version of the app when the plugin redirects you to that version on a later bundled app it will actually use the old versions of the plugin. I'll be looking into a fix for that tomorrow but in the meantime you'll want to make sure to hard delete the app from the device before installing to kill the old cache.

4. Lastly I think the version of your app currently deployed to `Master` has some build issues with the `vendor.js` when I did steps 1-3 and deployed the commit before that one your app successfully updated.

**Comment by diegozuber** [comment]
Thanks a lot @nphyatt 

1) The error with the gray screen is now fixed with cordova-plugin-ionic-webview v2.0.2

2) I did that exactly like this. The problem I have here is as follows. I deleted the plugins folder to recreate the Ionic Pro Build. Then I added a platform with `cordova platform add ios`. All plugins where loaded correctly, but the cordova-plugin-splashscreen changed automaticly in the `package.json` from
`"cordova-plugin-splashscreen": "git+https://github.com/nphyatt/cordova-plugin-splashscreen.git"`  
back to
`"cordova-plugin-splashscreen": "^5.0.2"` and loaded the official one instead of your forked one.
Is there something I did wrong?

3) Okay will do that.

4) This is very strange. Since that commit you're talking about I get on each commit a white screen. I tried to downgrade all dependency versions to the last working commit, but it didn't make any difference. If I deploy the commit before that, it works fine. Do you have any idea?

**Comment by diegozuber** [comment]
Some more testing info about my build problem over Ionic Pro(4) @nphyatt 
Maybe I need to create another issue for that, but I think it's somehow connected to the Ionic Deploy Plugin.

If I package the app on Ionic Deploy it works fine initially, without downloading any update. (I removed the auto update function on app start) But as soon as I make a Live Deploy Update to one of my recent commits, I get just a white screen. On android & iOS. On both platforms I'm getting this error in the console:
![bildschirmfoto 2018-08-02 um 10 26 54 am](https://user-images.githubusercontent.com/21689209/43575133-52765df4-9646-11e8-9aba-430875742473.png)

Strange is, that I'm not getting this error on the commit `a7f3ca` and before, just if you need to check my commits for that. I changed all code back to that commit, but the problems stays.I pushed my last working commit on the 30th of Juli at 22:45 CEST.

Let me now if I can help somehow with that problem.

**Comment by diegozuber** [comment]
@nphyatt I did some further testing and found out that the deployment of the commits stopped working after I updated `@ionic-native` from `v4.9.0` to `v4.11.0`. Could that be an issue with `cordova-plugin-ionic` or is that related to something else?

**Comment by diegozuber** [comment]
Check #116 

I created a new issue for that.

#107 is fixed in my case.
